infrastructure/src/main/java/org/corfudb/infrastructure/ICompactionTriggerPolicy.java

```diff
@@ -0,0 +1,16 @@
+package org.corfudb.infrastructure;
+
+import org.corfudb.runtime.CorfuRuntime;
+
+public interface ICompactionTriggerPolicy {
+    /**
+     * Computes if the compaction jvm should be triggered or not.
+     *
+     * @return
+     */
+    boolean shouldTrigger(long interval);
+
+    void markCompactionCycleStart();
+
+    void setCorfuRuntime(CorfuRuntime corfuRuntime);
```

Comment:
don't use setters in interfaces

Reply:
The corfuRuntime needs to be accessed only after it is initialized. Setting it at a later time through start ensures this. The corfuRuntime in CompactorService is a singletonResource which is not initialized completely until start().